src/UI/RealUserInterface.java

```diff
@@ -45,19 +42,20 @@ private Move getUserInput() {
         }
     }
 
-    private Move convertUserInputToEnum(String move) {
-        Move returnedMove = null;
+    private UserChoice convertUserInputToEnum(String move) {
+        UserChoice userInput = null;
+        UserChoice userChoice = new UserChoice(null, null);
```

Comment:
Try to avoid modifying the userChoice.
You can use your MoveChoice and Exit classes